src/core/src/package_managers/YumPackageManager.py

```diff
@@ -345,28 +370,434 @@ def get_package_size(self, output):
+            self.composite_logger.log_debug("Successfully disabled auto OS updates")
+
+        except Exception as error:
+            self.composite_logger.log_error("Could not disable auto OS updates. [Error={0}]".format(repr(error)))
+            raise
+
+    def disable_auto_os_update_for_yum_cron(self):
+        """ Disables auto OS updates, using yum cron service, and logs the default settings the machine comes with """
+        self.composite_logger.log("Disabling auto OS updates using yum-cron")
+        self.__init_auto_update_for_yum_cron()
+
+        self.backup_image_default_patch_configuration_if_not_exists()
+        self.composite_logger.log_debug("Preemptively disabling auto OS updates using yum-cron")
+        self.update_os_patch_configuration_sub_setting(self.download_updates_identifier_text, "no", self.yum_cron_config_pattern_match_text)
+        self.update_os_patch_configuration_sub_setting(self.apply_updates_identifier_text, "no", self.yum_cron_config_pattern_match_text)
+        self.disable_auto_update_on_reboot(self.yum_cron_disable_on_reboot_cmd)
+
+        self.composite_logger.log("Successfully disabled auto OS updates using yum-cron")
+
+    def disable_auto_os_update_for_dnf_automatic(self):
+        """ Disables auto OS updates, using dnf-automatic service, and logs the default settings the machine comes with """
+        self.composite_logger.log("Disabling auto OS updates using dnf-automatic")
+        self.__init_auto_update_for_dnf_automatic()
+
+        self.backup_image_default_patch_configuration_if_not_exists()
+        self.composite_logger.log_debug("Preemptively disabling auto OS updates using dnf-automatic")
+        self.update_os_patch_configuration_sub_setting(self.download_updates_identifier_text, "no", self.dnf_automatic_config_pattern_match_text)
+        self.update_os_patch_configuration_sub_setting(self.apply_updates_identifier_text, "no", self.dnf_automatic_config_pattern_match_text)
+        self.disable_auto_update_on_reboot(self.dnf_automatic_disable_on_reboot_cmd)
+
+        self.composite_logger.log("Successfully disabled auto OS updates using dnf-automatic")
+
+    def disable_auto_os_update_for_packagekit(self):
+        """ Disables auto OS updates, using packagekit service, and logs the default settings the machine comes with """
+        self.composite_logger.log("Disabling auto OS updates using packagekit")
+        self.__init_auto_update_for_packagekit()
+
+        self.backup_image_default_patch_configuration_if_not_exists()
+        self.composite_logger.log_debug("Preemptively disabling auto OS updates using packagekit")
+        #todo: uncomment after finding the correct value
+        # self.update_os_patch_configuration_sub_setting(self.download_updates_identifier_text, "false", self.dnf_automatic_config_pattern_match_text)
+        self.update_os_patch_configuration_sub_setting(self.apply_updates_identifier_text, "false", self.dnf_automatic_config_pattern_match_text)
```

Comment:
This is using dnf instead of packagekit - intentional?

Reply:
Resolved